azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -129,30 +130,33 @@ public ExecutorManager(final Props azkProps, final ExecutorLoader executorLoader
     this.updaterStage = updaterStage;
     this.executionFinalizer = executionFinalizer;
     this.updaterThread = updaterThread;
-    this.setupExecutors();
-    this.loadRunningExecutions();
-
-    this.queuedFlows = new QueuedExecutions(
-        azkProps.getLong(Constants.ConfigurationKeys.WEBSERVER_QUEUE_SIZE, 100000));
-
-    // The default threshold is set to 30 for now, in case some users are affected. We may
-    // decrease this number in future, to better prevent DDos attacks.
     this.maxConcurrentRunsOneFlow = azkProps
-        .getInt(Constants.ConfigurationKeys.MAX_CONCURRENT_RUNS_ONEFLOW,
+        .getInt(ConfigurationKeys.MAX_CONCURRENT_RUNS_ONEFLOW,
             DEFAULT_MAX_ONCURRENT_RUNS_ONEFLOW);
-    this.loadQueuedFlows();
-
-    this.cacheDir = new File(azkProps.getString("cache.directory", "cache"));
-
-    if (isMultiExecutorMode()) {
-      setupMultiExecutorMode();
-    }
-
     final long executionLogsRetentionMs =
         azkProps.getLong("execution.logs.retention.ms",
             DEFAULT_EXECUTION_LOGS_RETENTION_MS);
-
     this.cleanerThread = new CleanerThread(executionLogsRetentionMs);
+    this.executorInfoRefresherService = createExecutorInfoRefresherService();
+  }
+
+  void initialize() throws ExecutorManagerException {
+    if (initialized) {
+      return;
+    }
+    initialized = true;
+    this.setupExecutors();
```

Comment:
Moved this away from the constructor to fix AzkabanSingleServer (azkaban-solo). Because in AzkabanSingleServer Guice provisioning first creates instances of ExecutorManager etc., then starts the executor servlet, and finally web servlet. Executors can't be loaded (in ExecutorManager of azkaban-web) before the executor servlet has been started.